src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs

```diff
@@ -94,6 +103,23 @@ private async Task<string> GetRegionAsync(ICoreLogger logger)
             }
         }
 
+        private async void UpdateImdsUriApiVersionAsync(ICoreLogger logger)
+        {
+            Uri imdsErrorUri = new Uri("http://169.254.169.254/metadata/instance");
+            HttpResponse response = await _httpManager.SendGetAsync(imdsErrorUri, Headers, logger).ConfigureAwait(false);
+
+            LocalImdsErrorResponse errorResponse = JsonHelper.DeserializeFromJson<LocalImdsErrorResponse>(response.Body);
+            ImdsUri = buildImdsUri(errorResponse.NewestVersion[0]);
```

Comment:
What is the point of updating this private member instead of retrying with a new URI? If success, RegionDiscoveryProvider will never be called again anyway right?